cmd/kubent/main.go

```diff
@@ -154,10 +154,11 @@ func main() {
 		log.Fatal().Err(err).Str("name", "Rego").Msg("Failed to filter results")
 	}
 
-	printer, err := printer.NewPrinter(config.Output)
+	printer, err := printer.NewPrinter(config.Output, config.OutputFile)
 	if err != nil {
 		log.Fatal().Err(err).Msg("Failed to create printer")
 	}
+	defer printer.Close()
```

Comment:
this should happen inside of printer

Reply:
I don't know how to. Go doesn't have a concept of destructors, and `defer Close()` kind of a standard, see for example `io.Closer` [^1], and it's users responsibility to call it.

[^1]: https://pkg.go.dev/io#Closer